### ISSUE TITLE
[v1.0.0b] - Add three dots button for ingredients (and use it to display the menu)

### ISSUE THREAD

**Issue by simgunz** [mealie-recipes/mealie#1186]
### First Check

- [X] This is not a feature request
- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Mealie documentation, with the integrated search.
- [X] I already read the docs and didn't find an answer.

### What is the issue you are experiencing?

This is an unsability issue.

For each ingredient there should be a "three dots" button. The context menu should appear when clicking the three dots button. The drag button should only be dedicated to dragging the ingredient.

Basically implement what is already in place for the instruction steps.

![Screenshot_20220501_185145](https://user-images.githubusercontent.com/466270/166155973-a175f2ca-115f-4098-881c-8c939ea1f64c.jpg)
Image displaying the context menu and drag in the same button

### Deployment

Docker (Linux)

**Comment by hay-kot** [comment]
I don't want to have both a drag icon and a three dot icon. Would it be better to have just the three dots in this case?

**Comment by simgunz** [comment]
I think that showing the handlebar is good to give a hint to the user. Hidden features must always be avoided. I agree though that as it is now does not look too good.

I took some time to perform a UX analysis and I've proposed a first solution we can discuss.

**Problem analysis:**
![analysis_desktop_current](https://user-images.githubusercontent.com/466270/167349984-c2eca361-abd8-4957-8aa5-1791489b9130.jpg)

![analysis_mobile_current](https://user-images.githubusercontent.com/466270/167349989-d807ce66-054b-44e5-b264-ad334a648917.jpg)

**Proposed solution:**
![proposal_desktop](https://user-images.githubusercontent.com/466270/167350005-ec94788a-47e4-4a2f-a488-16c474f6d4ac.jpg)

![proposal_mobile](https://user-images.githubusercontent.com/466270/167350009-bab1c504-d1ad-4f26-85a1-9b88e29bd344.jpg)

**Comment by simgunz** [comment]
I've written "it is now less visually 'soft'", I meant is now "visually 'soft'", which mean is not as heavy as having drag handle and three dots next to each other.

**Comment by hay-kot** [comment]
I appreciate the really thought out feedback. This sort of detail makes it a lot easier to make changes, so thank you. 

I made some changes to the ingredient editor that reflect your ideas. However, I didn't make any changes for mobile. Making those changes would effectively require two different components of a significant amount of code to be duplicated. While not ideal, I think it's at least clear and usable. I do think the row feels a bit cluttered, but I'm not sure what else to do there without moving delete into the dropdown which I think harms usability. 

<img width="1017" alt="CleanShot 2022-05-09 at 13 45 16@2x" src="https://user-images.githubusercontent.com/64056131/167503891-104173f0-706c-4835-abea-9a81ca8807fc.png">

I also made some changes to the steps section as you described with some small changes. I moved the preview button into the menu (I'm not sure it's being used all that much anyways). And moved the placement of the dragger to where the trash used to be. I think your suggest has too much wasted/white-space.

<img width="664" alt="CleanShot 2022-05-09 at 13 47 18@2x" src="https://user-images.githubusercontent.com/64056131/167504274-40b3956b-3211-44a5-a6e1-fb0082277466.png">
